alternator/base64.cc

```diff
@@ -32,11 +32,11 @@
 // and the character used in base64 encoding to represent it.
 static class base64_chars {
 public:
-    static constexpr const char* to =
+    static constexpr const char to[] =
             "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
     int8_t from[255];
     base64_chars() {
-        static_assert(strlen(to) == 64);
+        static_assert(sizeof(to) == 64 + 1);
```

Comment:
Looks good. I don't understand your statement that strlen() can't be constexpr - the compilers are allowing more and more stuff to run in constexpr functions, and there is no reason not to have a constexpr version of strlen() that will work on constexpr strings. But I agree, if it's not implemented in clang, let's not use it, and you saw there is an easy workaround.

Reply:
I think that in case of g++ it's not even that `strlen` has a constexpr variant, but rather the whole call is optimized out and replaced with a compile-time known constant.

Reply:
Yes. g++ allowing it is a bug.